### PR TITLE
Add PyDM recipe.

### PR DESCRIPTION
**Pull request by hhslepicka**
Checklist

- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vend other packages
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/pydm/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{% set name = "pydm" %}
+{% set version = "1.7.3" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/slaclab/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: 0274ad8f906116bc03245478585ad596c7b89994a99ea3cd73f990efef6ecaf6
```
Comment (scopatz) [inline recipes/pydm/meta.yaml]: Can this be `noarch: python`?

File: recipes/pydm/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    - qtpy
+
+test:
+  imports:
+    - pydm
+
+about:
+  home: https://github.com/slaclab/pydm 
+  license: SLAC Open
+  license_file: LICENSE.md
```
Comment (scopatz) [inline recipes/pydm/meta.yaml]: `license_family: OTHER` should be added

File: recipes/pydm/meta.yaml (hunk 1 above)
Comment (hhslepicka) [reply recipes/pydm/meta.yaml]: I think so... Would this be a problem for `pyqt`?
I'm happy to change it...

Comment (scopatz) [reply recipes/pydm/meta.yaml]: I don't think so.  If you are not compiling anything in this recipe, then it can be noarch python, generally speaking.

Comment (hhslepicka) [reply recipes/pydm/meta.yaml]: Awesome... I changed it to `noarch: python`

File: recipes/pydm/meta.yaml (hunk 2 above)
Comment (hhslepicka) [reply recipes/pydm/meta.yaml]: Addressed with https://github.com/conda-forge/staged-recipes/pull/9639/commits/ea39b6b59d9794be6a30a3f04391e1cd766be7cb

**Comment by scopatz** [comment]
Excellent! Thanks!

**Pull request merged**